### DISCUSSION TITLE
Are there plans to support the type aware rules in typescript-eslint?

### DISCUSSION THREAD

**Discussion by jwhitmarsh** [oxc-project/oxc#7370] [General]
As per the title. I can see @Boshen has commented in various places that the decision was to not support them, but I can't find any discussions on what led to that decision and what the future looks like for oxc and type-aware rules.

**Comment by camc314** [comment]
See https://github.com/oxc-project/oxc/issues/3105

tldr no, using typescript is too slow, building our own typechecker is out of scope (at least at the moment)

**Reply by reneleonhardt** [reply]
typescript-eslint is mentioned 600 times in this codebase, does your reply mean there is a great deal of compatibility to typescript-eslint rules except any type checked rules?

https://www.joshuakgoldberg.com/blog/rust-based-javascript-linters-fast-but-no-typed-linting-right-now/

> **No Rust-based linter has integrated with TypeScript’s type checking APIs yet.**

https://voidzero.dev/posts/announcing-oxlint-1-stable

> including TypeScript-specific rules from typescript-eslint **(excluding type checked rules)**

**Reply by camc314** [reply]
We have a lot of coverage against the non-type aware `typescript-eslint` rules, however for rules that require type information (think no-floating-promises), we do not support yet. Also see #3105

**Reply by reneleonhardt** [reply]
Thank you, glad to see there are some efforts trying to achieve this.

As I understand it, LintPlugins are only a static set of built-in plugins remodeled after the most popular ESLint plugins.
Though there is no possibility for the community to rebuild plugins like eslint-comments outside of oxc? 
https://github.com/PostHog/posthog/pull/33714#issuecomment-2975937631

> * simple-import-sort - we have to fall back to ESLint's/Oxlint's built in sort-imports, where the major downside is the lack of an autofix yet – this is the biggest downside of the whole migration IMO, though still seems worth the speed gains (and I'm confident the Oxlint team and community will make this better too)
> * unused-imports - replaced by built-in no-unused-vars rule
> * react-google-translate - this did potentially flag real issues
> * storybook - sad to lose, but nothing critical
> * eslint-comments - nice-to-have
> * compat - seems this was only used for writing custom rules, which we won't be doing now
> * cypress - we're moving away from Cypress anyway

**Reply by camc314** [reply]
> simple-import-sort 

this will be built into the formatter (maybe not v1), but import sorting is planned

We're currently working on custom JS plugins, so the other rules should be able to be implemented (all be it slightly slower), in JS

**Reply by reneleonhardt** [reply]
Sounds good! But a LintPlugin API for Rust isn't planned, got it.

**Reply by camc314** [reply]
A rust side plugin system isn't planned at the moment. but we may explore it in future.

Yep the start is going to be JS plugins first, then explore from there
